ambassador/ambassador/ir/irauth.py

```diff
@@ -79,52 +77,19 @@ def add_mappings(self, ir: 'IR', aconf: Config):
                 if not self.cluster.merge(cluster):
                     self.post_error(RichStatus.fromError("auth canary %s can only change service!" % cluster.name))
                     cluster_good = False
-                    # self.ir.logger.debug("BAD MERGE %s" % cluster.as_json())
-                # else:
-                #     self.ir.logger.debug("GOOD MERGE %s" % cluster.as_json())
             else:
                 self.cluster = cluster
-                # self.ir.logger.debug("SET CLUSTER %s" % cluster.as_json())
 
         if cluster_good:
-            # self.ir.logger.debug("GOOD CLUSTER %s" % self.cluster.as_json())
             ir.add_cluster(self.cluster)
             self.referenced_by(self.cluster)
 
-        #     urls = []
-        #     protocols = {}
-        #
-        #     for svc in sorted(cluster_hosts.keys()):
-        #         _, tls_context = cluster_hosts[svc]
-        #
-        #         (svc, url, originate_tls, otls_name) = self.service_tls_check(svc, tls_context, host_rewrite)
-        #
-        #         if originate_tls:
-        #             protocols['https'] = True
-        #         else:
-        #             protocols['http'] = True
-        #
-        #         if otls_name:
-        #             filter_config['cluster'] = cluster_name + "_" + otls_name
-        #             cluster_name = filter_config['cluster']
-        #
-        #         urls.append(url)
-        #
-        #     if len(protocols.keys()) != 1:
-        #         raise Exception("auth config cannot try to use both HTTP and HTTPS")
-        #
-        #     self.add_intermediate_cluster(first_source, cluster_name,
-        #                                   'extauth', urls,
-        #                                   type="strict_dns", lb_type="round_robin",
-        #                                   originate_tls=originate_tls, host_rewrite=host_rewrite)
-        #
-        # name = "internal_%s_probe_mapping" % name
 
     def _load_auth(self, module: Resource):
         if self.location == '--internal--':
             self.sourced_by(module)
 
-        for key in [ 'path_prefix', 'timeout_ms', 'cluster', 'auth_service' ]:
+        for key in [ 'apiVersion', 'path_prefix', 'timeout_ms', 'cluster', 'auth_service' ]:
```

Comment:
It's probably OK to canary an `AuthService` using `ambassador/v0` with one using `ambassador/v1`, right?

Reply:
Yes, as far as the config is correct for the specific version, but then I assume Ambassador to fail even before even get to the IR.

Reply:
Oh ok, I see what you mean. Let me look into it.